src/Microsoft.AspNet.Identity/UserManager.cs

```diff
@@ -1063,10 +1062,9 @@ public virtual Task<IdentityResult> RemoveClaimAsync(TUser user, Claim claim)
                 throw new ArgumentNullException("roles");
             }
 
-            var userRoles = await userRoleStore.GetRolesAsync(user, CancellationToken);
             foreach (var role in roles)
```

Comment:
Should we do `roles.Distinct()` here to prevent duplicate roles from being added? I haven't looked very deep in `IsInRoleAsync()` but I assume it is possible it won't return true until things are saved to the database.